pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercuryOrderDao.java

```diff
@@ -207,18 +218,17 @@ public boolean orderNumberExists(String orderNumber) {
                     } else {
                         dbVals.resultValue = false;
                     }
-
-                } catch (Exception e) {
-                    throw new RuntimeException("Error getting values from db", e);
+                } catch (SQLException exception){
+                    dbVals.resultException = exception;
                 }
-            } catch (SQLException ex) {
+            } catch (Exception ex) {
                 dbVals.resultException = ex;
             }
             return dbVals;
         });
 
         if (results.resultException != null) {
-            throw new RuntimeException("Error checking if values exist in db", results.resultException);
+            throw new DsmInternalError("Error checking if values exist in db", results.resultException);
```

Comment:
This is a very generic message that does not help with debugging. A message saying that you could not find an Mercury order number in the DB, which includes the order number, would be better.